internal/engine/synclet_build_and_deployer.go

```diff
@@ -161,28 +186,37 @@ func (sbd *SyncletBuildAndDeployer) PostProcessBuild(ctx context.Context, result
 		return
 	}
 
-	if _, ok := sbd.getDeployInfoForImage(result.Image); !ok {
-		deployInfo, err := sbd.getDeployInfo(ctx, result.Image)
-		if err != nil {
-			// There's a variety of reasons why we might not be able to get the deploy info.
-			// The cluster could be in a transient bad state, or the pod
-			// could be in a crash loop because the user wrote some code that
-			// segfaults. Don't worry too much about it, we'll fall back to an image build.
-			logger.Get(ctx).Debugf("failed to get deployInfo: %v", err)
-			return
-		}
-		sbd.setDeployInfoForImage(result.Image, deployInfo)
+	info, ok := sbd.deployInfoForImageOrNew(result.Image)
+	if ok {
+		// This info was already in the map, nothing to do.
+		return
+	}
+
+	// We just made this info, so populate it.
+	err := sbd.populateDeployInfo(ctx, result.Image, info)
+	if err != nil {
+		// There's a variety of reasons why we might not be able to get the deploy info.
+		// The cluster could be in a transient bad state, or the pod
+		// could be in a crash loop because the user wrote some code that
+		// segfaults. Don't worry too much about it, we'll fall back to an image build.
+		logger.Get(ctx).Debugf("failed to get deployInfo: %v", err)
+		return
 	}
 }
 
-func (sbd *SyncletBuildAndDeployer) getDeployInfo(ctx context.Context, image reference.NamedTagged) (DeployInfo, error) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncletBuildAndDeployer-getDeployInfo")
+func (sbd *SyncletBuildAndDeployer) populateDeployInfo(ctx context.Context, image reference.NamedTagged, info *DeployInfo) error {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncletBuildAndDeployer-populateDeployInfo")
 	defer span.Finish()
 
+	if info == nil {
+		info = newEmptyDeployInfo()
+	}
+	defer info.markReady()
```

Comment:
If we hit an error here, are we marking an unpopulated `deployInfo` as ready? (and then will that stay in the map forever?)

Reply:
yyyyes we do. good catch.